File: mobile/interfaces/Consultas/TipoConsultaHorarios.ts
import { HorarioConsultas } from "./HorarioConsultas";
import { TipoConsultas } from "./TipoConsultas";

export interface TipoConsultaHorarios {
    tipoConsultaId: number;
    tipoConsulta: TipoConsultas;
    horarioId: number;
    horario: HorarioConsultas
    reservado: boolean;
    data: Date;
}

File: backend/EsteticaDominio/Consulta/TipoConsultaHorarios.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EsteticaDominio.Consulta
{
    public class TipoConsultaHorarios
    {
        public int TipoConsultaId { get; set; }
        public TipoConsulta TipoConsulta { get; set; }
        public int HorarioId { get; set; }
        public HorarioConsultas Horario { get; set; }
        public bool Reservado { get; set; }
        public DateTime DataReserva { get; set; }
    }
}